ogc/src/main/scala/geotrellis/server/ogc/utils/ExpressionUtils.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.server.ogc.utils
+
+import geotrellis.raster.TargetCell
+import com.azavea.maml.ast.Expression
+import cats.syntax.option._
+
+object ExpressionUtils {
+  def bindExpression(expr: Expression, fun: Expression => Expression): Expression = {
+    def deepMap(expression: Expression, f: Expression => Expression): Expression =
+      f(expression).withChildren(expression.children.map(f).map(deepMap(_, f)))
+
+    deepMap(expr, fun)
+  }
```

Comment:
It looks like maml lacks some convenient function to traverse the expression tree.